framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -1022,6 +1022,37 @@ protected int getCacheSize() {
         }
     }
 
+    /**
+     * Returns an AnnotatedTypeMirror representing the annotated type of {@code elt} as a type
+     * declaration.
+     *
+     * <p>It is not possible to tell whether an element is a use of a type or a declaration, so this
+     * method differentiates the two.
+     *
+     * @param elt element whose type is requested
+     * @return the annotated type of {@code elt} as a type declaration
+     */
+    public AnnotatedTypeMirror getAnnotatedTypeAsTypeDecl(Element elt) {
+        return getAnnotatedType(elt);
```

Comment:
Can you add a comment why this is the right implementation?
Why not move the `isTypeDecl` field here and document that subclasses should use that field to implement special logic?